core/src/test/java/org/mskcc/oncotree/utils/OncotreeUtilTestConfig.java

```diff
@@ -0,0 +1,826 @@
+        Map<String, TumorType> expectedTumorTypeMap = new HashMap<>();
+        String[] rawTestValueSource = getRawTestValueSource();
+        final int valuesPerCase = 5;
+        if (rawTestValueSource.length % valuesPerCase != 0) {
+            throw new Exception("Error : malformed rawTestValueSource");
+        }
+        final int caseCount = rawTestValueSource.length / valuesPerCase;
+        if (caseCount < 1) {
+            throw new Exception("Error : no test cases defined in rawTestValueSource");
+        }
+        expectedTumorTypeMap = new HashMap<>(caseCount + 1);
+        for (int pos = 0; pos < rawTestValueSource.length; pos = pos + valuesPerCase) {
+            TumorType nextType = new TumorType();
+            String code = rawTestValueSource[pos];
+            if (code != null && code.trim().length() > 0) {
+                nextType.setCode(code.trim());
+            } else {
+                fail("Error : list of expected oncotree codes contains a node with no code");
+            }
+            String name = rawTestValueSource[pos + 1];
+            if (name != null && name.trim().length() > 0) {
+                nextType.setName(name.trim());
+            }
+            MainType nextTypeMainType = new MainType();
+            String mainType = rawTestValueSource[pos + 2];
+            if (mainType != null && mainType.trim().length() > 0) {
+                nextTypeMainType.setName(mainType.trim());
+            }
+            nextType.setMainType(nextTypeMainType);
+            String color = rawTestValueSource[pos + 3];
+            if (color != null && color.trim().length() > 0) {
+                nextType.setColor(color.trim());
+            }
+            String parentCode = rawTestValueSource[pos + 4];
+            if (parentCode != null && parentCode.trim().length() > 0) {
+                nextType.setParent(parentCode.trim());
+            }
+            expectedTumorTypeMap.put(code, nextType);
+        }
+        return expectedTumorTypeMap;        
+    }
+    
+    
+    private List<MainType> setupExpectedMainTypeList() throws Exception {
+        Set<MainType> expectedMainTypeSet = new HashSet<>();
+        List<MainType> expectedMainTypeList = new ArrayList<>();
+        String[] rawTestValueSource = getRawTestValueSource();
+        final int valuesPerCase = 5;
+        if (rawTestValueSource.length % valuesPerCase != 0) {
+            throw new Exception("Error : malformed rawTestValueSource");
+        }
+        final int caseCount = rawTestValueSource.length / valuesPerCase;
+        if (caseCount < 1) {
+            throw new Exception("Error : no test cases defined in rawTestValueSource");
+        }
+        for (int pos = 0; pos < rawTestValueSource.length; pos = pos + valuesPerCase) {
+            MainType nextType = new MainType();
+            String mainType = rawTestValueSource[pos + 2];
+            if (mainType != null && mainType.trim().length() > 0) {
```

Comment:
also .. add a check here for rawTestValueSource[pos + 4] != null ... to skip the "TISSUE" node (and other top level nodes if we add detached cell line trees later)